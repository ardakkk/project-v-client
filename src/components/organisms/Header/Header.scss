.Header {
    &_title {
        @media screen and (max-width: 479px) {
            display: none;
        }
    }

    &_search {
        width: 100%;
        position: relative;
        margin-left: 0;
        margin-right: 16px;
        border-radius: 4px;
        background-color: rgba(255, 255, 255, 0.15);
        cursor: text;
        align-self: center;
        transition: all 0.3s ease;

        &___focused {
            background-color: rgba(255, 255, 255, 0.25);
        }
    }

    &_searchIcon {
        width: 56px;
        height: 100%;
        position: absolute;
        display: flex;
        justify-content: center;
        align-items: center;
    }

    &_input {
        position: relative;
        display: inline-flex;
        align-items: center;

        input {
            width: 100%;
            padding: 8px 8px 8px 56px;
            transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
            color: #fff;
        }
    }

    &_menu {
        display: flex;
        justify-content: flex-end;
    }
}
