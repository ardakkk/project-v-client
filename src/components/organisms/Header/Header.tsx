import * as React from 'react';

import { AppBar, Toolbar, Typography } from '@material-ui/core';
import Badge from '@material-ui/core/Badge';
import Grid from '@material-ui/core/Grid';
import Hidden from '@material-ui/core/Hidden';
import IconButton from '@material-ui/core/IconButton';
import InputBase from '@material-ui/core/InputBase';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import AccountCircle from '@material-ui/icons/AccountCircle';
import MailIcon from '@material-ui/icons/Mail';
import MenuIcon from '@material-ui/icons/Menu';
import MoreIcon from '@material-ui/icons/MoreVert';
import NotificationsIcon from '@material-ui/icons/Notifications';
import SearchIcon from '@material-ui/icons/Search';

import * as classes from './Header.scss';

export const Header: React.FunctionComponent<any> = props => {
    const [isFocused, setFocus] = React.useState<boolean>(false);
    const [mobileMoreAnchorEl, setMobileMoreAnchorEl] = React.useState(null);

    const onFocus = React.useCallback(() => {
        setFocus(true);
    }, []);
    const onBlur = React.useCallback(() => {
        setFocus(false);
    }, []);

    const isMobuleMenuOpenMenu: boolean = Boolean(mobileMoreAnchorEl);
    const handleMobileMenuOpen = (event: any): void => {
        setMobileMoreAnchorEl(event.currentTarget);
    };
    const handleMobileMenuClose = (): void => {
        setMobileMoreAnchorEl(null);
    };

    const renderMobileMenu = (
        <Menu
            anchorEl={mobileMoreAnchorEl}
            anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
            keepMounted={true}
            transformOrigin={{ vertical: 'top', horizontal: 'right' }}
            open={isMobuleMenuOpenMenu}
            onClose={handleMobileMenuClose}>
            <MenuItem>
                <IconButton aria-label='show 4 new mails' color='inherit'>
                    <Badge badgeContent={4} color='secondary'>
                        <MailIcon />
                    </Badge>
                </IconButton>
                <p>Messages</p>
            </MenuItem>
            <MenuItem>
                <IconButton
                    aria-label='show 11 new notifications'
                    color='inherit'>
                    <Badge badgeContent={11} color='secondary'>
                        <NotificationsIcon />
                    </Badge>
                </IconButton>
                <p>Notifications</p>
            </MenuItem>
            <MenuItem>
                <IconButton
                    aria-label='account of current user'
                    aria-controls='primary-search-account-menu'
                    aria-haspopup='true'
                    color='inherit'>
                    <AccountCircle />
                </IconButton>
                <p>Profile</p>
            </MenuItem>
        </Menu>
    );

    return (
        <Grid className={classes.Header}>
            <AppBar>
                <Toolbar>
                    <Grid container={true}>
                        <Hidden smUp={true}>
                            <Grid item={true} xs={2}>
                                <IconButton
                                    edge='start'
                                    className={classes.menuButton}
                                    color='inherit'
                                    aria-label='open drawer'>
                                    <MenuIcon />
                                </IconButton>
                            </Grid>
                        </Hidden>
                        <Grid item={true} lg={2}>
                            <Hidden mdDown={true}>
                                <Typography variant='h6' noWrap={true}>
                                    Project V
                                </Typography>
                            </Hidden>
                        </Grid>
                        <Grid
                            item={true}
                            lg={7}
                            xs={8}
                            className={`${classes.Header_search} ${
                                isFocused ? classes.Header___search_focused : ''
                            }`}>
                            <div className={classes.Header_searchIcon}>
                                <SearchIcon />
                            </div>
                            <div className={classes.Header_input}>
                                <InputBase
                                    placeholder='Searchâ€¦'
                                    inputProps={{ 'aria-label': 'search' }}
                                    onFocus={onFocus}
                                    onBlur={onBlur}
                                />
                            </div>
                        </Grid>
                        <Hidden smUp={true}>
                            <Grid
                                item={true}
                                xs={2}
                                className={classes.Header_menu}>
                                <IconButton
                                    aria-label='show more'
                                    aria-haspopup='true'
                                    color='inherit'
                                    onClick={handleMobileMenuOpen}>
                                    <MoreIcon />
                                </IconButton>
                            </Grid>
                        </Hidden>
                    </Grid>
                </Toolbar>
            </AppBar>
            {renderMobileMenu}
        </Grid>
    );
};
