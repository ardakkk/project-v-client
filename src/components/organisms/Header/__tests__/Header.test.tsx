import { shallow } from 'enzyme';
import * as React from 'react';
import { Header } from '../Header';

describe('Header organism', () => {
    it('Snapshot should match', () => {
        const component = shallow(<Header />);

        expect(component).toMatchSnapshot();
    });
});
