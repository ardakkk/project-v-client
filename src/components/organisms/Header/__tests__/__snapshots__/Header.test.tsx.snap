// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header organism Snapshot should match 1`] = `
<WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(AppBar))>
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
      >
        <Hidden
          smUp={true}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={2}
          >
            <WithStyles(ForwardRef(IconButton))
              aria-label="open drawer"
              color="inherit"
              edge="start"
              onClick={[Function]}
            >
              <MenuIcon />
            </WithStyles(ForwardRef(IconButton))>
          </WithStyles(ForwardRef(Grid))>
        </Hidden>
        <WithStyles(ForwardRef(Grid))
          item={true}
          lg={2}
        >
          <Hidden
            mdDown={true}
          >
            <WithStyles(ForwardRef(Typography))
              noWrap={true}
              variant="h6"
            >
              Project V
            </WithStyles(ForwardRef(Typography))>
          </Hidden>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="undefined "
          item={true}
          lg={7}
          xs={8}
        >
          <div>
            <SearchIcon />
          </div>
          <div>
            <WithStyles(ForwardRef(InputBase))
              inputProps={
                Object {
                  "aria-label": "search",
                }
              }
              onBlur={[Function]}
              onFocus={[Function]}
              placeholder="Quick Find"
            />
          </div>
        </WithStyles(ForwardRef(Grid))>
        <Hidden
          smDown={true}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            lg={3}
          >
            <WithStyles(ForwardRef(IconButton))
              color="inherit"
            >
              <AddIcon />
            </WithStyles(ForwardRef(IconButton))>
            <WithStyles(ForwardRef(IconButton))
              color="inherit"
            >
              <Brightness5Icon />
            </WithStyles(ForwardRef(IconButton))>
            <WithStyles(ForwardRef(IconButton))
              color="inherit"
            >
              <WithStyles(ForwardRef(Badge))
                badgeContent={11}
                color="secondary"
              >
                <NotificationsIcon />
              </WithStyles(ForwardRef(Badge))>
            </WithStyles(ForwardRef(IconButton))>
            <WithStyles(ForwardRef(IconButton))
              color="inherit"
            >
              <SettingsIcon />
            </WithStyles(ForwardRef(IconButton))>
          </WithStyles(ForwardRef(Grid))>
        </Hidden>
        <Hidden
          smUp={true}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={2}
          >
            <WithStyles(ForwardRef(IconButton))
              aria-haspopup="true"
              aria-label="show more"
              color="inherit"
              onClick={[Function]}
            >
              <MoreVertIcon />
            </WithStyles(ForwardRef(IconButton))>
          </WithStyles(ForwardRef(Grid))>
        </Hidden>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <Component
    handleMobileMenuClose={[Function]}
    mobileMoreAnchorEl={null}
  />
  <Component
    isOpenLeftSideBar={false}
    toggleLeftSideBarDrawer={[Function]}
  />
</WithStyles(ForwardRef(Grid))>
`;
